streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -95,9 +69,7 @@ public interface StateStore {
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
-    default void init(final StateStoreContext context, final StateStore root) {
-        init(StoreToProcessorContextAdapter.adapt(context), root);
-    }
+    default void init(final StateStoreContext context, final StateStore root) {}
```

Comment:
This is used to replace the deprecated `init`, so I guess it should NOT have default implementation after removing the deprecated version? 

@guozhangwang WDYT?